extension/js/common/composer.ts

```diff
@@ -1201,6 +1208,45 @@ export class Composer {
         if (method === 'forward') {
           this.urlParams.subject = 'Fwd: ' + this.urlParams.subject;
           await this.retrieveDecryptAddForwardedMsg(determined.lastMsgId);
+        } else {
+          try {
+            const { raw } = await Google.gmail.msgGet(this.urlParams.acctEmail, this.urlParams.threadMsgId, 'raw');
+            const msgMimeContent = await Mime.process(Buf.fromBase64UrlStr(raw!));
+            if (!msgMimeContent.from || !msgMimeContent.headers.date || !msgMimeContent.blocks || !msgMimeContent.blocks.length) {
+              return;
+            }
+            if (!this.urlParams.draftId) {
+              const blocksToDisplay = msgMimeContent.blocks.filter(b => b.type === 'encryptedMsg' ||
+                b.type === 'plainText' || b.type === 'plainHtml');
+              const contentToDisplay: Array<string> = [];
```

Comment:
All of this below looks good, but it's somewhat duplicated. Very similar functionality is in `retrieveDecryptAddForwardedMsg`, so these two should be merged together and put into one function (the pulling and decrypting of the message, which just returns plain text).

I think your method does it slightly better, because you are cycling through all of the blocks with `Mime.process`, so you'll be sure to catch the needed content.

On the other hand, you are pulling the raw message. If it contains a 15 mb attachment, that may take a while. So you may need to add a loading % sign, maybe next to the three dots, eg "Loading previous message quote... 18%". That means the three dots should probably be visible while this happens, but have opacity 50% and clicking it should not do anything (maybe should also have `cursor: default`). Then when it loads, opacity of (...) goes to 1 and it can be clicked if user wants to.

Either that, or adopt the faster but less precise method from forwarding messages, which may just be good enough. I don't know, your call - you can test it on `flowcrypt.compatibility` emails to see how many of them (that are not marked with a `fix` label) load without issues. You should be able to load almost any plain, plain html, encrypted, or signed message.